siliconcompiler/package.py

```diff
@@ -0,0 +1,247 @@
+import os
+import sys
+import re
+import shutil
+import siliconcompiler
+
+class Sup:
+    '''SiliconCompiler Unified Package (SUP) class.
+
+    Main object used to interact with the SiliconCompiler
+    Package Management System API.
+
+    '''
+
+    def __init__(self, registry=None):
+
+        self.chip = siliconcompiler.Chip()
+
+        # Local cache location
+        if 'SC_HOME' in os.environ:
+            home = os.environ['SC_HOME']
+        else:
+            home = os.environ['HOME']
+
+        self.cache = os.path.join(home,'.sc','registry')
```

Comment:
Can we automatically create cache here if it doesn't exist? When I first ran the example it crashed due to not being able to find registry.

Reply:
Yes, you are right, this is a bug that needs fixing.